src/server/pps/server/master.go

```diff
@@ -106,29 +113,55 @@ func (a *apiServer) master() {
 	})
 }
 
+func (a *apiServer) setPipelineFailure(ctx context.Context, pipelineInfo *pps.PipelineInfo) error {
+	// Set pipeline state to running
+	_, err := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		pipelineName := pipelineInfo.Pipeline.Name
+		pipelines := a.pipelines.ReadWrite(stm)
+		pipelineInfo := new(pps.PipelineInfo)
+		if err := pipelines.Get(pipelineName, pipelineInfo); err != nil {
+			return err
+		}
+		pipelineInfo.State = pps.PipelineState_PIPELINE_FAILURE
+		pipelines.Put(pipelineName, pipelineInfo)
+		return nil
+	})
+	return err
+}
+
```

Comment:
Will pipelines that are in failure state automatically recover and get out of failure state? Or is there some action the user needs to take?